src/jllvm/vm/Interpreter.cpp

```diff
@@ -450,6 +450,23 @@ std::uint64_t jllvm::Interpreter::executeMethod(const Method& method, std::uint1
                 context.push<std::uint32_t>(array->size());
                 return NextPC{};
             },
+            [&](CheckCast checkCast)
+            {
+                auto* object = context.pop<ObjectInterface*>();
+                context.push(object);
+                if (!object)
+                {
+                    return NextPC{};
+                }
+
+                ClassObject* classObject = getClassObject(classFile, checkCast.index);
```

Comment:
Is the optimization like in the JIT not worth it here?

Reply:
I am assuming you're referring to https://github.com/JLLVM/JLLVM/issues/244 here?

Optimizations such as those are more likely to be effective in the implementation of `Object::instanceOf`rather than directly within the interpreter. The implementation already does what is essentially an upcast check. Changing the order in which the subclass and interface checks are performed could potentially change average speeds although that would certainly have to be measured.

I think a more suitable optimization for the interpreter could be to have an LRU cache for these checks as the are likely to be repeated checks for the same types in sequence.

Id prefer to keep any non-obvious optimizations to future PRs tho

Reply:
I was actually thinking of using `forNameLoaded` and instead.

Reply:
Ahhh sorry, I wasn't sure. 

Yes that makes sense, lemme change that.